### PR TITLE
Creating neural network for final boss comparison

### PR DESCRIPTION
**Pull request by megan-pa**
## Type of Change
- [X] Feature
- [ ] Bugfix
- [ ] Both
- [ ] Other (please specify)

## What's been done
- Created a neural network with hyper parameter tuning to approximate player behaviour clusters generated by the GMM
- Started a Jupyter notebook with evaluation into the neural network including loss curves and batch size training 
- Made changes to the player tactic model based on the new clusters found using the GMM

## How it's been done
- Trained a multi-layer perceptron making use of the Sequential model provided by TensorFlow based on generated clusters
- Applied standardisation (scaler mean/scale) into the training pipeline 
- Performed hyper parameter tuning to find the optimal batch size and epochs
- Added early stopping and dropout to evaluate neural network performance and automate epochs number selection
- Performed 5-fold cross-validation on the neural network to evaluate generalisation performance 

## What's next
- Implement "NeuralNetworkModel.cs" to load the model into the game
- Integrate neural network into the game and compare performance against the GMM

## Related Issues
- #205

### CODE REVIEW

**Review by pabloapca — APPROVED** [review]
AMAZING, GREAT ADDITION

**Review by ethanr04 — APPROVED** [review]
Amazing work!
